src/cli/utils/utils.go

```diff
@@ -243,12 +243,36 @@ func GetStorageProviderAndOverrides(
 		return provider, nil, clues.Stack(err)
 	}
 
-	overrides := map[string]string{}
-
 	switch provider {
 	case storage.ProviderS3:
-		overrides = flags.S3FlagOverrides(cmd)
+		return provider, flags.S3FlagOverrides(cmd), nil
+	case storage.ProviderFilesystem:
+		return provider, flags.FilesystemFlagOverrides(cmd), nil
+	}
+
+	return provider, nil, clues.New("unknown storage provider: " + provider.String())
+}
+
+func MakeAbsoluteFilePath(p string) (string, error) {
```

Comment:
Seems like this has some opinions about what `p` should be.  Could you please add a func header describing expected i/o for different variations of p, so that future maintainers don't need to read the func exhaustively to understand its behavior?